### PR TITLE
[server][clients] Provide IC provider to schema reader

### PR DESCRIPTION
**Pull request by majisourav99**
## Summary, imperative, start upper case, don't end with a period

Currently for request to schema reader in venice server does not provide any Invocation context which leads to exceptions from 
downstream client calls. This PR adds invocation context to schema reader clients calls in venice server.

## How was this PR tested?

CI build
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java [hunk 1]
```diff
@@ -124,14 +124,16 @@ public DaVinciBackend(
 
       SchemaReader partitionStateSchemaReader = ClientFactory.getSchemaReader(
           ClientConfig.cloneConfig(clientConfig)
-              .setStoreName(AvroProtocolDefinition.PARTITION_STATE.getSystemStoreName()));
```
Comment (sixpluszero) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java:127]: I have a general question - I don't see in this diff how the passed in IC invoked? Could you explain how it is working under the new implementation?

Comment (majisourav99) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java:127]: AFAIK IC provider is passed implicity when client is called from user applications. But when venice-server calls client, only then we need to set those up.

File: clients/venice-thin-client/src/test/java/com/linkedin/venice/client/schema/RouterBackedSchemaReaderTest.java [hunk 2]
```diff
@@ -250,7 +250,8 @@ public void testGetLatestValueSchemaWithSupersetSchema() throws Exception {
     try (SchemaReader schemaReader = new RouterBackedSchemaReader(
         () -> clientMock,
         Optional.empty(),
-        Optional.of(schema -> schema.toString().equals(valueSchemaStr1)))) {
+        Optional.of(schema -> schema.toString().equals(valueSchemaStr1)),
+        null)) {
```
[the remark this answers is not in the document]
Comment (majisourav99) [reply clients/venice-thin-client/src/test/java/com/linkedin/venice/client/schema/RouterBackedSchemaReaderTest.java]: I updated with a mock

**Review by sixpluszero — DISMISSED** [review]
Looks good to me, thanks!

**Review by sixpluszero — APPROVED** [review]

**Pull request merged**